### ISSUE TITLE
Programmatically add label to Decision Log

### ISSUE THREAD

**Issue by PietervdWerk** [open-policy-agent/opa#7131]
## What is the underlying problem you're trying to solve?
When extending OPA with your own functions or plugins, there is nothing in the decision log indicating that the OPA runtime that ran the evaluation is in fact a custom build OPA runtime. 

## Describe the ideal solution
I would like to programmatically add a label to the decision log that indicates the custom version. A ldflag can be used to pass the build tag into the binary. And some kind of hook could be used to add the label to manager's labels.

```go
package main

import (
	"os"

        "github.com/open-policy-agent/opa/config"
	"github.com/open-policy-agent/opa/cmd"
	"gitlab.com/n-id/core/pkg/utilities/log/v2"
)

var version = "dev"

func main() {
        config.AddLabel("build-tag", version)
        // Plugin and extensions code

	if err := cmd.RootCommand.Execute(); err != nil {
		log.Errorln(err)
		os.Exit(1)
	}
}
```

## Describe a "Good Enough" solution
I do not know a simpler solution, but all ideas are welcome!

## Additional Context
The OPA config allows a labels setting to add labels to the Decision Log for that instance, as far as I could see, there is no way to add to the config programmatically.

https://openpolicyagent.slack.com/archives/CBR63TK2A/p1729599666570109

**Comment by charlieegan3** [comment]
Hi, thanks for raising the issue here. ([Slack Thread](https://openpolicyagent.slack.com/archives/CBR63TK2A/p1729599666570109))

This might be a bit of a hack, but could you use `os.Setenv("OPA_BUILD", "my-custom-build")`

and then use environment variables in the config: https://www.openpolicyagent.org/docs/latest/configuration/#using-environment-variables-in-configuration

for the `labels` key: https://www.openpolicyagent.org/docs/latest/configuration/#miscellaneous

I no it's not super neat, but it might unblock you.

**Comment by charlieegan3** [comment]
Ok interesting, so beyond the most minimal config file needed to download the bundle, what configuration is being set by clients in their OPA config files? Are they making use of other OPA features that are out of your control?

I was wondering if you might be able to use Discovery Bundles https://www.openpolicyagent.org/docs/latest/management-discovery/ to minimise the different things that users need to change. The realised configuration might be able to abstract the variable substitution?

**Comment by charlieegan3** [comment]
Hey @PietervdWerk, have you had a chance to test out the above Discovery Bundle suggestion to see if it'd work for your project?

**Comment by PietervdWerk** [comment]
Hi yes, this is working for now. Would be a nice feature still, quite a bit of overhead like this, but it works.

**Comment by charlieegan3** [comment]
That's great! Do you think we need to keep this issue open? It'd be great to get some more details about your use case to help weigh this up. What is it about the Discovery Bundles route that introduces the overhead for you? Lots of OPAs to manage? Requires new deployments to 'hard to reach' places? or something else? 🤔

**Comment by PietervdWerk** [comment]
We didn't end up using discovery bundles. The decisionlog already has a revision which we can link back to a tag which was build using a specific version of the runtime. It would have reduced the steps needed to figure out that information, which could help our support engineers in the future, but it is not a problem yet, and we aim to be fully backwards compatible so no weird logs should happen anyway.

I did try out the discovery bundle method and it was quite pleasant to work with, and a good choice should we want to implement this. We didn't want to introduce a second bundle that (if ill configured) could take down all clients using the discovery, since it was a nice to have anyway.

We can close this issue :)

**Comment by charlieegan3** [comment]
Ok thanks for the extra detail @PietervdWerk! I'll close this one for now
